### PR TITLE
Alexander/isom 1156 create add new block drawer state

### PR DESCRIPTION
**Pull request by alexanderleegs**
## Problem
This PR introduces the block drawer state. The popover state for each block is currently WIP, pending design - the text will be added in a future PR.

Screenshot 2024-06-18 at 12 06 08 PM

Reference: [figma](https://www.figma.com/design/g7LnLn4IUcUC9dYxgsHEJM/IsomerCMS-V2-(working-title%3A-%22Isomer-Studio%22)?node-id=5727-31436&m=dev)

### CODE REVIEW

File: apps/studio/src/components/PageEditor/Editor.tsx [hunk 1]
```diff
@@ -61,32 +62,36 @@ const renderers: JsonFormsRendererRegistryEntry[] = [
...
+    //       <JsonForms
+    //         schema={jsonSchema}
+    //         data={editorValue}
+    //         renderers={renderers}
+    //         cells={materialCells}
+    //         onChange={({ data }) => onChange(data)}
+    //       />
+    //     </Box>
+    //   </ThemeProvider>
+    // </Box>
```
Comment (seaerchin) [inline apps/studio/src/components/PageEditor/Editor.tsx:94]: nit: can remove

File: apps/studio/src/components/PageEditor/Editor.tsx [hunk 2]
```diff
@@ -61,32 +62,36 @@ const renderers: JsonFormsRendererRegistryEntry[] = [
...
-            data={editorValue}
-            renderers={renderers}
-            cells={materialCells}
-            onChange={({ data }) => onChange(data)}
-          />
-        </Box>
-      </ThemeProvider>
-    </Box>
+    <ComponentSelector
+      onClose={() => console.log('close')}
```
Comment (seaerchin) [inline apps/studio/src/components/PageEditor/Editor.tsx:66]: nit: toast is clearer for an indication that `onClose` is firing

**Review by seaerchin — COMMENTED** [review]
how can i test this ah

**Comment by alexanderleegs** [comment]
> how can i test this ah

Can be tested at `/editor`! It's a temporary page that should be removed later

**Review by seaerchin — APPROVED** [review]
Looking good! I think what worked well for me in this PR was having the `BlockSelector` abstraction, which made the overall component alot easier to understand

File: apps/studio/src/components/PageEditor/ComponentSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+              sectionType="divider"
+              description="TODO"
+            />
+          </BlockList>
+        </Section>
+        <Section>
+          <SectionTitle title="External Content" />
+          <BlockList>
+            <BlockItem
+              label="Youtube"
```
Comment (harishv7) [inline apps/studio/src/components/PageEditor/ComponentSelector.tsx:219]: nit: YouTube

Comment (alexanderleegs) [reply apps/studio/src/components/PageEditor/ComponentSelector.tsx]: changed in https://github.com/opengovsg/isomer/pull/187/commits/71703b3913474b61c929c028c71dea8f069681c4

File: apps/studio/src/components/PageEditor/Editor.tsx (hunk 1 above)
Comment (alexanderleegs) [reply apps/studio/src/components/PageEditor/Editor.tsx]: Removed in https://github.com/opengovsg/isomer/pull/187/commits/71703b3913474b61c929c028c71dea8f069681c4

File: apps/studio/src/components/PageEditor/Editor.tsx (hunk 2 above)
Comment (alexanderleegs) [reply apps/studio/src/components/PageEditor/Editor.tsx:66]: Leaving this for now since we'll be replacing it shortly!

**Pull request merged**